pytest-embedded-serial/pytest_embedded_serial/serial.py

```diff
@@ -52,12 +54,26 @@ def __init__(
         self._meta = meta
 
         self.port = port
+        self.port_location = port_location
         self.baud = baud
 
-        if port is None:
-            raise ValueError('Please specify port')
-
-        if not isinstance(port, str):
+        if port is None and not port_location:
+            raise ValueError('Please specify port or provide the port location')
+
+        if port_location:
+            is_port_found = False
+            for port in list_ports.comports():
+                if port.location == port_location:
+                    if self.port and port.device != self.port:
+                        is_port_found = False
+                        break
+                    self.port = port.device
+                    is_port_found = True
+                    break
+            if not is_port_found:
+                raise ValueError('The specified port cannot be found in the provided location.')
```

Comment:
the error message should be more friendly. and you could use the for...else... to simplify it a bit.

```python
        if port_location:
            for port in list_ports.comports():
                if port.location == port_location:
                    if self.port and port.device != self.port:
                        raise ValueError(f'The specified location {port_location} binds with port {port.device}, not {self.port}')

                    self.port = port.device
                    break
            else:
                raise ValueError(f'The specified location {port_location} cannot be found.')
```